lib/wibox/widget/slider.lua

```diff
@@ -505,6 +507,45 @@ function slider:fit(_, width, height)
     return width, height
 end
 
+function slider:layout(context, width, height)
+    local handle_height, handle_width = height, self._private.handle_width
+        or beautiful.slider_handle_width
+        or math.floor(height / 2)
+    local handle_widget = self._private.handle_widget
+        or nil
+    if handle_widget then
+        local x, y = 0, 0
+        local w, h
+        local value = self._private.value or self._private.min or 0
+        local min, _, interval = get_extremums(self)
+        w, h = base.fit_widget(self, context, handle_widget, handle_width, handle_height)
+        local margins = self._private.handle_margins
+            or beautiful.slider_handle_margins
+
+        local x_offset, y_offset = 0, 0
+
+        if margins then
+            if type(margins) == "number" then
```

Comment:
Please create a custom setter for `set_handle_margins` to always convert the margins to a table at the setter level. This is a mini API change, so please add a note in the `behavior changes` section of `doc/89-NEWS.md`.

You can then remove the type code from `:draw()`. This way we wont have duplicated type checking code. You will also have less tests to write since that `if` will be gone.

like
```lua
function slider:set_handle_margins(value)
    if type(value) == "number" then
        value = {
            top    = value,
            bottom = value,
            left   = value,
            right  = value,
         }
    end

    self._private.handle_margins = value
    self:emit_signal( "property::handle_margins", value)
    self:emit_signal( "widget::redraw_needed" )
end
```